ape_hardhat/providers.py

```diff
@@ -522,7 +522,7 @@ def build_command(self) -> List[str]:
 
         return cmd
 
-    def reset_fork(self):
+    def reset_fork(self, new_block_number: int):
```

Comment:
This should default to `None` and the coerce to `self.fork_block_number` to preserve existing breaking and prevent this from counting as a breaking change.

Reply:
i added another commit and included your suggestions